documentation/docs/25-build-and-deploy/40-adapter-node.md

```diff
@@ -162,6 +170,58 @@ MY_CUSTOM_ORIGIN=https://my.site \
 node build
 ```
 
+> [Socket activation](#socket-activation) does not work when a prefix is specified because `LISTEN_PID` and `LISTEN_FDS` are set by the OS and cannot be renamed.
+
+## Graceful shutdown
+
+By default `adapter-node` gracefully shuts down the HTTP server when a `SIGTERM` or `SIGINT` signal is received. It does this by calling [`server.close`](https://nodejs.org/api/http.html#serverclosecallback), [`server.closeIdleConnections`](https://nodejs.org/api/http.html#servercloseidleconnections) and [`server.closeAllConnections`](https://nodejs.org/api/http.html#servercloseallconnections) under the hood. This will
+
+1. wait for requests that have already been made but not received a response yet to finish
+2. reject new requests
+3. close connections once they become idle
+4. and finally, close any remaining connections that are still active after [`SHUTDOWN_TIMEOUT`](#shutdown_timeout) seconds.
+
+> If you want to customize this behaviour you can use a [custom server](#custom-server).
+
+## Socket activation
+
+Most Linux operating systems today use a modern process manager called systemd to start the server and run and manage services. You can configure your server to allocate a socket and start and scale your app on demand. This is called [socket activation](http://0pointer.de/blog/projects/socket-activated-containers.html). In this case, the OS will pass two environment variables to your app — `LISTEN_PID` and `LISTEN_FDS`. The adapter will verify that these variables are correct and then listen on file descriptor 3 which refers to a systemd socket unit that you will have to create.
+
+To take advantage of socket activation follow these steps.
+
+1. Run your app as a [systemd service](https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html). It can either run directly on the host system or inside a container (using Docker or a systemd portable service for example).
+
+```ini
+/// file: /etc/systemd/system/myapp.service
+[Service]
+Environment=NODE_ENV=production
+ExecStart=/usr/bin/node /usr/bin/myapp/build
+```
+
+2. Create an accompanying [socket unit](https://www.freedesktop.org/software/systemd/man/latest/systemd.socket.html). The adapter only accepts a single socket.
+
+```ini
+/// file: /etc/systemd/system/myapp.socket
+[Socket]
+ListenStream=3000
+
+[Install]
+WantedBy=sockets.target
+```
+
+3. Make sure systemd has recognised both units by running `sudo systemctl daemon-reload`. Then enable the socket on boot and start it immediately using `sudo systemctl enable --now myapp.socket`.
+
+The app will then automatically start once the first request is made to `localhost:3000`. Additionally, if you pass an `IDLE_TIMEOUT` environment variable to your app the adapter will terminate it after receiving no requests for `IDLE_TIMEOUT` seconds.
```

Comment:
I think everything after "Additionally" could be moved up to the `IDLE_TIMEOUT` section above. It seems like an option that might not be too heavily used and this section is getting a bit long.

Reply:
I think `IDLE_TIMEOUT` will probably almost always be used when socket activation is enabled because one of the main benefits of socket activation is that you can automatically start/stop the app based on usage. But I'll try to clean/shorten it a bit.

Reply:
One way to condense it a bit would be to demonstrate `IDLE_TIMEOUT` in bullet 1. By putting it here we end up duplicating the code block, which I think makes it longer.

Reply:
Okay, I just updated the docs.